ui/app/models/pki/action.js

```diff
@@ -0,0 +1,181 @@
+import Model, { attr } from '@ember-data/model';
+import { inject as service } from '@ember/service';
+import lazyCapabilities, { apiPath } from 'vault/macros/lazy-capabilities';
+import { withFormFields } from 'vault/decorators/model-form-fields';
+import { withModelValidations } from 'vault/decorators/model-validations';
+
+const validations = {
+  type: [{ type: 'presence', message: 'Type is required.' }],
+  commonName: [{ type: 'presence', message: 'Common name is required.' }],
+  issuerName: [
+    {
+      validator(model) {
+        if (model.issuerName === 'default') return false;
+        return true;
+      },
+      message: 'Issuer name must be unique across all issuers and not be the reserved value default.',
+    },
+  ],
+};
+
+@withModelValidations(validations)
+@withFormFields()
+export default class PkiActionModel extends Model {
+  @service secretMountPath;
+
+  /* formType import */
+  @attr('string') pemBundle;
+
+  /* formType generate-root */
+  @attr('string', {
+    possibleValues: ['exported', 'internal', 'existing', 'kms'],
+    noDefault: true,
+  })
+  type;
+
+  @attr('string') issuerName; // REQUIRED, cannot be "default"
+
+  @attr('string') keyName; // cannot be "default"
+
+  @attr('string', {
+    defaultValue: 'default',
+    label: 'Key reference',
+  })
+  keyRef; // type=existing only
+
+  @attr('string') commonName; // REQUIRED
+
+  @attr('string', {
+    label: 'Subject Alternative Names (SANs)',
+  })
+  altNames; // comma sep strings
+
+  @attr('string', {
+    label: 'IP Subject Alternative Names (IP SANs)',
+  })
+  ipSans;
+
+  @attr('string', {
+    label: 'URI Subject Alternative Names (URI SANs)',
+  })
+  uriSans;
+
+  @attr('string', {
+    label: 'Other SANs',
+  })
+  otherSans;
+
+  @attr('string', {
+    defaultValue: 'pem',
+    possibleValues: ['pem', 'der', 'pem_bundle'],
+  })
+  format;
+
+  @attr('string', {
+    defaultValue: 'der',
+    possibleValues: ['der', 'pkcs8'],
+  })
+  privateKeyFormat;
+
+  @attr('string', {
+    defaultValue: 'rsa',
```

Comment:
Oh, I see. My thinking with this section was that, since key info in this form is under a toggle the UI should default to the backend's default rather than forcing the user to open up that section and make a selection. I'll check with Ivana if that's the intended functionality though.